src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/AddExtendedQueryTagError.sql

```diff
@@ -32,7 +32,7 @@ AS
             THROW 50404, 'Instance does not exist or has not been created.', 1;
 
         --Check if tag exists and in Adding status.
-        IF NOT EXISTS (SELECT * FROM dbo.ExtendedQueryTag WITH (HOLDLOCK) WHERE TagKey = @tagKey AND TagStatus = 0)
+        IF NOT EXISTS (SELECT * FROM dbo.ExtendedQueryTag WITH (UPDLOCK) WHERE TagKey = @tagKey AND TagStatus = 0)
```

Comment:
This makes sense. We can allow INSERTs to happen concurrently, as if the entry were added in the middle of our transaction we would fail anyway here without it.

Reply:
If I understand correctly, what you mean is that `UPDLOCK` doesn't block concurrent `INSERT`, but this procedure will fail here as no qualified tag, so it should be fine?